src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,23 @@
+import Popup from "./Popup";
+export default class PopupWithConfirm extends Popup {
+  constructor(popupSelector, handleDelete) {
+    super({ popupSelector });
+
+    this._popupSelector = popupSelector;
+    this._handleDelete = handleDelete;
+  }
+
+  setSubmitFunction(submitFunction) {
+    this._handleDelete = submitFunction;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this.trashModal = document.querySelector(`${this._popupSelector}`);
```

Comment:
`trashModal`

It's not a trash modal. It's `confirmationModal`


But... you have `this._popupElement` as the modal 


**No** need to search the **modal** again in the inheritors of class `Popup` because the parent has already found it in its constructor as `this._popupElement`. It’s available in any child.